### PR TITLE
patch for ip cve fix

### PR DESCRIPTION
**Pull request by JedML**
**Description**

- added patch for updating ip@2.0.0 to ip@2.0.1

**Motivation**

There is a CVE for ip@2.0.0 that is fixed with ip@2.0.1

*Issue #, if available:*

*Description of changes:*
created patch for this change, applied patches, then copied all to patched folder

*Testing*
- pulled fresh version of code - oss
- build code `yarn install --pure-lockfile --verbose`
- ran smoke tests to verify prior to adding any patches `yarn smoketest --web --browser chromium`
```
  70 passing (6m)
  8 pending

✨  Done in 514.59s.
```
- cleaned vscode dir
- pull fresh version from source `git submodule update --init`
- applied new patch `quilt push patches/security.diff`
- cd into vscode
- build code `yarn install --pure-lockfile --verbose`
- ran smoke tests with current patch `yarn smoketest --web --browser chromium`

```
  70 passing (6m)
  8 pending

✨  Done in 501.63s.
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: patched-vscode/yarn.lock [hunk 1]
```diff
@@ -5567,9 +5567,9 @@ invert-kv@^2.0.0:
   integrity sha512-wPVv/y/QQ/Uiirj/vh3oP+1Ww+AWehmi1g5fFWGPF6IpCBCDVrhgHRMvrLfdYcwDh3QJbGXDW4JAuzxElLSqKA==
 
 ip@^2.0.0:
-  version "2.0.0"
-  resolved "https://registry.yarnpkg.com/ip/-/ip-2.0.0.tgz#4cf4ab182fee2314c75ede1276f8c80b479936da"
-  integrity sha512-WKa+XuLG1A1R0UWhl2+1XQSi+fZWMsYKffMZTTYsiZaUD8k2yDAj5atimTUD2TZkyCkNEeYE5NhFZmupOGtjYQ==
+  version "2.0.1"
```
Comment (JedML) [inline patched-vscode/yarn.lock:5570]: this is the fix based off of https://github.com/microsoft/vscode/pull/205768/files

File: patched-vscode/src/vs/workbench/browser/media/code-icon.svg
Comment (arkaprava08) [inline patched-vscode/src/vs/workbench/browser/media/code-icon.svg:1]: Why are these getting deleted. All the icons seems to be getting deleted

File: patches/ip-cve.diff
Comment (arkaprava08) [inline patches/ip-cve.diff:1]: I would recommend the diff file to be just cve and then in the diff file we can add ip as one of the cve fixes

File: patches/series [hunk 2]
```diff
@@ -1,8 +1,9 @@
+ip-cve.diff
```
Comment (arkaprava08) [inline patches/series:1]: same here, rename this to like security-patches.diff

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -53,6 +53,22 @@ opensource-codeofconduct@amazon.com with any additional questions or comments.
...
+
+0. You can go through the patch stack with `quilt push` and `quilt pop`.
+1. Create a new patch (`quilt new {name}.diff`) or use an existing patch.
+1. Add the file(s) you are patching (`quilt add [-P patch] {file}`). A file
+   **must** be added before you make changes to it.
+1. Make your changes. Patches do not need to be independent of each other but
+   each patch must result in a working sagemaker-code-editor without any broken in-between
+   states otherwise they are difficult to test and modify.
+1. Add your changes to the patch (`quilt refresh`)
+1. Add a comment in the patch about the reason for the patch and how to
```
Comment (arkaprava08) [inline CONTRIBUTING.md:68]: The  numbering seems off. Is this correct?

File: patches/series [hunk 4]
```diff
@@ -1,8 +1,9 @@
+security.diff
```
Comment (arkaprava08) [inline patches/series:1]: nit: how about security-patches.diff or security-fixes.diff

**Review by arkaprava08 — COMMENTED** [review]
Please update the commit message to something like Bumping ip from x.y.z to a.b.c for cve: <cveid or link or n/a>

File: CONTRIBUTING.md (hunk 3 above)
Comment (JedML) [reply CONTRIBUTING.md:68]: numbering is markdown and is correct

**Comment by JedML** [comment]
@arkaprava08 please note that the current commit message is:
```
patch for ip cve fix

**Description**

- added patch for updating ip@2.0.0 to ip@2.0.1

**Motivation**

There is a CVE for ip@2.0.0 that is fixed with ip@2.0.1
```

Seems to have all of the requested data

**Review by arkaprava08 — APPROVED** [review]
LGTM

**Pull request merged**